spaceship/lib/spaceship/connect_api/models/device.rb

```diff
@@ -54,8 +54,23 @@ def self.all(client: nil, filter: {}, includes: nil, limit: nil, sort: nil)
         return resps.flat_map(&:to_models)
       end
 
+      # @param client [ConnectAPI] ConnectAPI client.
+      # @param platform [String] The platform of the device.
+      # @param include_disabled [Bool] Whether to include disable devices. false by default.
+      # @return (Device) Find a device based on the UDID of the device. nil if no device was found.
+      def self.find_by_udid(device_udid, client: nil, platform: nil, include_disabled: false)
+        self.all(client: client).find do |device|
+          device.udid.casecmp(device_udid) == 0 && (include_disabled ? true : device.enabled?)
+        end
+      end
+
       def self.create(client: nil, name: nil, platform: nil, udid: nil)
         client ||= Spaceship::ConnectAPI
+
+        # Find the device by UDID and return it in case it already exists
+        existing = self.find_by_udid(udid, client: client, platform: platform)
+        return existing if existing
+
```

Comment:
I think this works but I'm a little bit unsure about making use an existing one 🤔  Especially if this is fixing an action.

I think this "find or create" should be handled in `register_device`. Either that or there should be a method on this model called `find_or_create` just so that is more clear 😊 

Thoughts?

Reply:
I agree! Single responsibility principle.

Right now we're using this method in 2 places, so instead of duplicating this logic in each of them, I'll create this `find_or_create` function as you suggested 😊

Reply:
Done! Lmk your thoughts 🤗